### PR TITLE
avoid crashes caused by bad DNSnames in gsql backends

### PR DESCRIPTION
**Pull request by mind04**

### CODE REVIEW

File: pdns/backends/gsql/gsqlbackend.cc [hunk 1]
```diff
@@ -1013,7 +1041,11 @@ bool GSQLBackend::get(DNSResourceRecord &r)
     } catch (SSqlException &e) {
       throw PDNSException("GSQLBackend get: "+e.txtReason());
     }
-    extractRecord(row, r);
+    try {
+      extractRecord(row, r);
+    } catch (...) {
+      goto skiprow;
```
Comment (Habbie) [inline pdns/backends/gsql/gsqlbackend.cc]: silent failure? shouldn't we SERVFAIL if we cannot understand what we are seeing?

Comment (mind04) [inline pdns/backends/gsql/gsqlbackend.cc]: Yes, but for now silent ignoring is the safest thing to do. I'm not sure  an empty dnsname exception is caught down the line and i'm not aware of an other method to trigger a servfail from a gsql backend without destroying the backend.

**Comment by cmouse** [comment]
Shouldn't we catch empty DNSName at least in dnswriter xfrName? It should refuse to move empty dnsname?

File: pdns/backends/gsql/gsqlbackend.cc (hunk 1 above)
Comment (Habbie) [inline pdns/backends/gsql/gsqlbackend.cc]: Ok, so the big question, strictly for alpha1, is "silently hide broken records" versus "kill backend on a broken record". I'll think about this.

Comment (mind04) [inline pdns/backends/gsql/gsqlbackend.cc]: For records the problem is theoretical. It is unlikely a backend will return a broken DNSName if you ask for a good one in a query. The real problem lies in broken zones tsigkeys and metadata. 
One broken dnsname in domains for example is  enough to kill the masterslavecomminicator in the current master. 

This pull is not intended as the ultimate solution. It is just buying us some time to fix things for real while people enjoy a more or less working alpha release.

Comment (Habbie) [inline pdns/backends/gsql/gsqlbackend.cc]: Yes, understood, like I said: 'strictly for alpha1'. I'm going to agree with the 'ignore them' policy for now, but we need to open a ticket to track this right after we merge it. Will wait for coverage of @cmouse's questions before merging.

**Pull request merged**